lib/phoenix/pubsub/pg2_adapter.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Phoenix.PubSub.PG2Adapter do
+  use GenServer
+
+  @moduledoc """
+  The server for the PG2Adapter
+  """
+
+  @private_pg2_group {:phx, :global}
+
+
+  def start_link(opts) do
+    GenServer.start_link __MODULE__, [], name: Dict.fetch!(opts, :name)
+  end
+
+  def init(_opts) do
+    {:ok, local_pid} = Phoenix.PubSub.Local.start_link()
+    :ok = :pg2.create(@private_pg2_group)
+    :ok = :pg2.join(@private_pg2_group, self)
+    {:ok, %{local_pid: local_pid}}
+  end
+
+  def handle_call({:subscribe, pid, topic}, _from, state) do
```

Comment:
Maybe one idea to avoid blocking this too much is to support "redirect" in `Phoenix.PubSub`. Basically, if you return `{:redirect, pid_or_name}`, we will do the same call from `Phoenix.PubSub` but to the given pid and topic.